qrexec/tools/qrexec_policy_daemon.py

```diff
@@ -99,6 +99,78 @@ async def handle_client_connection(log, policy_cache,
     finally:
         writer.close()
 
+async def handle_qrexec_connection(log, policy_cache,
+                                   reader, writer):
+
+    args = {}
+
+    try:
+        untrusted_data = await reader.read(65536)
+        if len(untrusted_data) > 65535:
+            log.error('Request length too long: %d', len(data))
+            return
+        try:
+            # Qrexec guarantees that this will be present
+            service_descriptor_len = untrusted_data.index(b'\0')
+            # This has already been validated by qrexec, so at least parts of
+            # it can be trusted.
+            service_descriptor = untrusted_data[:service_descriptor_len]
+            qrexec_command_with_arg = service_descriptor[:service_descriptor.index(b' ')]
+            qrexec_command_len = qrexec_command_with_arg.find(b'+')
+            if qrexec_command_len <= 0:
+                if qrexec_command_len:
+                    log.warning('No service specified in policy query')
+                else:
+                    log.error('Invoked via an empty service name?')
+                return
+            # The service we are being queried for
+            service_queried = qrexec_command_with_arg[qrexec_command_len + 1:]
+            if not service_queried:
+                log.warning('Empty string is not a valid service')
+
+            ### SANITIZE BEGIN
+            untrusted_data = untrusted_data[service_descriptor_len + 1:]
+
+            if len(untrusted_data) > 63:
+                log.warning('Request data too long: %d', len(untrusted_data))
+                return
+            split = untrusted_data.find(b'\0')
+            if not (1 <= split <= 31):
+                log.warning('Invalid data from qube')
+                return
+            untrusted_source = untrusted_data[:split].decode('ascii', 'strict')
+            untrusted_target = untrusted_data[split + 1:].decode('ascii', 'strict')
+            if not (1 <= len(untrusted_target) <= 31):
+                log.warning('Invalid data from qube')
+                return
+
+            # these throw exceptions if the domain name is not valid
+            utils.sanitize_service_name(qrexec_arg, True)
+            utils.sanitize_domain_name(untrusted_source, True)
+            utils.sanitize_domain_name(untrusted_target, True)
+            ### SANITIZE END
+            source, intended_target = untrusted_source, untrusted_target
+        except (ValueError, UnicodeError):
+            log.warning('Invalid data from qube')
+            return
+
+        result = await handle_request(
+                source=source,
+                intended_target=intended_target,
+                service_and_arg=service_queried.decode('ascii', 'strict'),
+                domain_id = -1,
+                process_ident = -1
+                assume_yes_for_ask=True,
```

Comment:
For the very case of gui-daemon and clipboard, this will be enough. But for generic case, I'd like to distinguish also ask from allow, and get more details: https://github.com/QubesOS/qubes-issues/issues/6264
Maybe leave it as is for now, but name the service `policy.EvalSimple` and have full `policy.Eval` later?